server/tests/api.test.js

```diff
@@ -4,37 +4,49 @@ import Database from "../database.js"
 
 const databaseDisabledError = { error: 'Database not implemented' };
 const incorrectQueryParamsError = { error: 'Incorrect query parameters' };
+console.assert(process.env.NODE_ENV == "test", "This test suite should only be run in the test environment");
 
 beforeAll(async () => {
-    process.env.DATABASE_ENABLED = true;
+    // process.env.DATABASE_ENABLED = true;
     await API.init();
+    await Database.init();
+
+    // would be good to do before each test
+    // but it's so slow and none of the tests are mutating anyway
+    // it would be good to make this not random, but Math.random has no seed function
+    await Database.loadTestData();
 });
 
 beforeEach(async () => {
-    await Database.init();
 })
 
 describe('API', () => {
-    it('Queries Not Needing Arguments', async () => {
-        const persistentEndpoints = [
-            '/api/agents',
-            '/api/bots',
-            '/api/count-game-pages',
-            '/api/agent-games',
-            '/api/top-winrate',
-            '/api/most-improved',
-            '/api/most-improving',
-        ];
+    describe('Queries Not Needing Arguments', () => {
+        Object.entries({
+            'agents': 10,
+            'bots': 1,
+            'most-improved': 9,
+            'most-improving': 9,
+            'top-winrate': 9, // TODO should this be 10?
```

Comment:
It doesn't seem to be, test was run quite a few times. If it is random, the test failing will let us know, but in either case it would be better to have a non-random test database by creating a good one, and exporting it